### PR TITLE
 Validations Khadri 04112024 0240 pm 

### PR DESCRIPTION
**Pull request by roohishaik128**
Validations For Product & Stock Module

### CODE REVIEW

File: SQL_Scripts/scripts.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+create schema 2024_batch_workshop_servlet_mobile_shop_app; 
+use 2024_batch_workshop_servlet_mobile_shop_app;
+create table product (ID int auto_increment,Name varchar(200),primary key (ID));
+create table stock(id int auto_increment primary key, type varchar(200), qty  int);
+create table mobile(product_brand varchar(50),product_price double,product_model varchar(50),arrived_date_time datetime);
+create table charger(product_brand varchar(50),product_price double,product_model varchar(50),arrived_date_time datetime);
+create table powerbank(product_brand varchar(50),product_price double,product_model varchar(50),arrived_date_time datetime);
+create table backcover(product_brand varchar(50),product_price double,product_model varchar(50),arrived_date_time datetime);
+create table headset(product_brand varchar(50),product_price double,product_model varchar(50),arrived_date_time datetime);
```
Comment (manvithagali52) [inline SQL_Scripts/scripts.sql]: why we need sql scripts it is already in master branch

File: src/main/java/com/khadri/jakarta/stock/dao/StockDao.java
Comment (manvithagali52) [inline src/main/java/com/khadri/jakarta/stock/dao/StockDao.java:1]: why you deleted this stockDao class

File: src/main/java/com/khadri/jakarta/stock/form/BackCoverForm.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	public Double getProductPrice() {
+		return productPrice;
+	}
+
+	public String getProductModel() {
+		return productModel;
+	}
+
+	public Date getArrivedDateTime() {
+		return arrivedDateTime;
```
Comment (manvithagali52) [inline src/main/java/com/khadri/jakarta/stock/form/BackCoverForm.java]: why your are adding all this files

File: src/main/java/com/khadri/jakarta/stock/servlet/modify/StockModifyServlet.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				stockDao.updateHeadSetPrice(productPrice, productModel);
+				stockDao.updateBackCoverPrice(productPrice, productModel);
+				pw.println("Product price modified successfully!");
+			} catch (Exception e) {
+				pw.println("Error: Could not modify the product price.");
+				e.printStackTrace();
+			}
+		}
+	}
+}
```
Comment (manvithagali52) [inline src/main/java/com/khadri/jakarta/stock/servlet/modify/StockModifyServlet.java]: Add only your validation files why you are adding all files

File: src/main/java/com/khadri/jakarta/stock/form/BackCoverForm.java
Comment (sgreddy123) [inline src/main/java/com/khadri/jakarta/stock/form/BackCoverForm.java:1]: you need to add only your validation files.

File: src/main/java/com/khadri/jakarta/product/servlet/modify/ProductModifySearchPageServlet.java [hunk 4]
```diff
@@ -36,20 +36,38 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
...
+		sb.append("<head>");
+		sb.append("<title>Modify Product Details</title>");
+		sb.append("<script type='text/javascript'>");
+		sb.append("function validateForm() {");
+		sb.append("var id = document.getElementById('productId').value;");
+		sb.append("if (id == '') {");
+		sb.append("alert('Please enter a Product ID ..!');");
+		sb.append("return false;");
+		sb.append("}");
+		sb.append("if(isNaN(productIdComponent)) { ");
```
Comment (sgreddy123) [inline src/main/java/com/khadri/jakarta/product/servlet/modify/ProductModifySearchPageServlet.java:49]: isNaN or isName?

Comment (roohishaik128) [reply src/main/java/com/khadri/jakarta/product/servlet/modify/ProductModifySearchPageServlet.java:49]: isNanN is method when productId given a alphabet It gives one alert

File: SQL_Scripts/scripts.sql (hunk 1 above)
Comment (roohishaik128) [reply SQL_Scripts/scripts.sql]: I added Wrongly

**Review by khadri2021 — APPROVED** [review]

**Pull request merged**